src/utils.ts

```diff
@@ -29,3 +33,93 @@ export function epochToBatchId(epoch: BigInt): BigInt {
 export function getBatchId(event: EthereumEvent): BigInt {
   return epochToBatchId(event.block.timestamp)
 }
+
+export function getOwlDecimals(): BigInt {
+  return OWL_DECIMALS
+}
+
+export function coalesce<T>(valueOpt: T | null, defaultValue: T): T {
+  if (valueOpt == null) {
+    return defaultValue
+  } else {
+    return valueOpt!
+  }
+}
+
+export function toWei(amount: BigInt, decimalsOpt: BigInt | null): BigInt {
+  let decimals = coalesce<BigInt>(decimalsOpt, DEFAULT_DECIMALS).toI32()
+  return amount.times(TEN.pow(u8(decimals)))
+}
+
+export function greatestCommonDenominator(numerator: BigInt, denominator: BigInt): BigInt {
+  if (denominator.isZero()) {
+    return numerator
+  } else {
+    return greatestCommonDenominator(denominator, numerator.mod(denominator))
+  }
+}
+
+export function reduce(numerator: BigInt, denominator: BigInt): BigInt[] {
+  let greatestCommonDenominator = greatestCommonDenominator(numerator, denominator)
+  return [numerator.div(greatestCommonDenominator), denominator.div(greatestCommonDenominator)]
+}
+
+/**
+ * Calculate the price given twe amounts of two tokens
+ *
+ * For the market Orange-Apple, we say:
+ *   - Orange is the base currency
+ *   - Apple is the quote currency
+ *   - And the price would be, how many Apples you pay for one Orange
+ *
+ *
+ * @param amountBase Amount in the base token
+ * @param decimalsBaseOpt
+ * @param amountQuote
+ * @param decimalsQuoteOpt
+ */
+export function calculatePrice(
+  amountBase: BigInt,
+  decimalsBaseOpt: BigInt | null,
+  amountQuote: BigInt,
+  decimalsQuoteOpt: BigInt | null,
+): BigInt[] {
+  let decimalsBase = coalesce<BigInt>(decimalsBaseOpt, DEFAULT_DECIMALS)
+  let decimalsQuote = coalesce<BigInt>(decimalsQuoteOpt, DEFAULT_DECIMALS)
+  let decimalsDifference = decimalsBase.minus(decimalsQuote)
+
+  // log.info('[utils:calculatePrice] Base: {} ({}) and Quote: {} ({})', [
+  //   amountBase.toString(),
+  //   decimalsBase.toString(),
+  //   amountQuote.toString(),
+  //   decimalsQuote.toString(),
+  // ])
```

Comment:
Before I get a comment of "dead code", I'm happy to remove this ones, BUT they are useful for debugging, and I don't want to have them for the normal deployment

Reply:
lol, too late hahaha